### ISSUE TITLE
Formatting with Binary Arguments Causing OOM Error

### ISSUE THREAD

**Issue by theonewolf** [redis/hiredis#120]
I am using Ubuntu 12.04 LTS, 64-bit, with libhiredis0.10 installed (and libhiredis-dev).  redis-server Version: 2:2.2.12-1build1

I setup this gist illustrating some very odd behavior:

https://gist.github.com/3298361

In trying to create a minimal error case I had to leave some oddities in:

(1) I only get OOM with the buffer at (`uint8_t buf[];`) having more than 1 byte inside it
(2) Along with the `for` loop in place
(3) and formatting my `SETEX` command with `%b %d %b`

If you:

(1) Set the buf to be one character only `uint8_t buf[1]` the failure doesn't occur
(2) Comment out the for loop, have the buf any size, the failure doesn't occur
(3) Make the `%b %d %b` formatting `%s %d %s`, the failure doesn't occur

Basically, I needed an extra variable on the stack and a for loop to make the issue appear with arguments formatted as binary.

Perhaps I have something messed up in my code?
